### ISSUE TITLE
search.word__link_count command eats next statement

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#306]
I have been running a number of builds over various wikis and have finally realised an issue.

in the middle of the build script for wikinews is:

    // calculate a score for each page using the page-to-page links
    add     ('en.wikinews.org' , 'search.page__page_score') {iteration_max = 100;}

    // update link score statistics for the search tables
    add     ('en.wikinews.org' , 'search.link__link_score') {page_rank_enabled = 'y';}

    // update word count statistics for the search_word table
    add     ('en.wikinews.org' , 'search.word__link_count')

    // cleanup all downloaded files as well as temporary files
    add     ('en.wikinews.org' , 'util.cleanup')          {delete_tmp = 'y'; delete_by_match('*.xml|*.sql|*.bz2|*.gz');}

    // v2 html generator; allows for multi-threaded / multi-machine builds
    add     ('en.wikinews.org' , 'wiki.mass_parse.init') {cfg {ns_ids = '0|4|14|8';}}

    add     ('en.wikinews.org' , 'wiki.mass_parse.exec') {
    ...

When I have run this script I wondered why files were not tidied up.
I then decided to rearrange slightly to (remove the cleanup)

    // calculate a score for each page using the page-to-page links
    add     ('en.wikinews.org' , 'search.page__page_score') {iteration_max = 100;}

    // update link score statistics for the search tables
    add     ('en.wikinews.org' , 'search.link__link_score') {page_rank_enabled = 'y';}

    // update word count statistics for the search_word table
    add     ('en.wikinews.org' , 'search.word__link_count')

    // v2 html generator; allows for multi-threaded / multi-machine builds
    add     ('en.wikinews.org' , 'wiki.mass_parse.init') {cfg {ns_ids = '0|4|14|8';}}

    add     ('en.wikinews.org' , 'wiki.mass_parse.exec') {
    ...

This time when I ran it I got an error telling me
`SQL error or missing database (no such table: xomp_page)`

I guessed this to be that the mass_parse.init did not happen.
That is, word__link_count is eating the next statement.

I took a look at some other no parameter commands eg

    add     ('en.wikinews.org' , 'wiki.imagelinks');
    add     ('en.wikinews.org' , 'wiki.page_link');

In reviewing the code, I noticed that neither Imglnk_bldr_cmd.java nor Pglnk_bldr_cmd.java have an override to the Invk function whereas Xobldr__word__link_count.java does.
I think this is the cause of the consumption

**Comment by gnosygnu** [comment]
As always, thanks for the detailed write-up.

I think the main problem is .gfs and the lack of detailed error-handling. Both the scripts above show a missing-semi-colon:

> add     ('en.wikinews.org' , 'search.word__link_count') // <---- there should be a semi-colon after the right-parenthesis

Since there is no terminating semi-colon, I believe the gfs parser will just run on and gobble the next statement (in this case, the wiki.mass_parse.init command). This would explain your observation that the mass_parse.init did not happen.

In retrospect, I should've not created a new gfs file-format, and just used json, but I really wanted comments and I disliked json's strict quoting-rules for keys. So, I wrote gfs, and now I ended up where I ended up. Oh well...

Regarding your other point on Imglnk_bldr_cmd and Pglnk_bldr_cmd: They both extend Xob_sql_dump_base so they will function normally, just like other commands. I really do believe that the consumption is caused by the missing semi-colon, but do let me know if I'm missing anything else.

Thanks again.
